service/src/test/java/bio/terra/cbas/controllers/VerifyPactsRunSetsApiController.java

```diff
@@ -65,20 +72,80 @@ void before(PactVerificationContext context) {
     context.setTarget(new MockMvcTestTarget(mockMvc));
   }
 
+  @State({"initialize dependencies"})
+  public void initialize() throws Exception {
+
+    // Arrange WDS
+    RecordResponse myRecordResponse = new RecordResponse();
+    myRecordResponse.setId("FOO1");
+    myRecordResponse.setType("FOO");
+    RecordAttributes myRecordAttributes = new RecordAttributes();
+    myRecordAttributes.put("foo_rating", 10);
+    myRecordAttributes.put("bar_string", "this is my bar_string");
+    myRecordResponse.setAttributes(myRecordAttributes);
+    when(wdsService.getRecord(any(), any())).thenReturn(myRecordResponse);
+
+    // Arrange methodVersion
+    UUID methodVersionUUID = UUID.fromString("90000000-0000-0000-0000-000000000009");
+    MethodVersion myMethodVersion =
+        new MethodVersion(
+            methodVersionUUID,
+            new Method(
+                UUID.fromString("00000000-0000-0000-0000-000000000009"),
+                "myMethod name",
+                "myMethod description",
+                OffsetDateTime.now(),
+                methodVersionUUID,
+                "myMethod source"),
+            "myMethodVersion name",
+            "myMethodVersion description",
+            OffsetDateTime.now(),
+            UUID.fromString("0e811493-6013-4fe7-b0eb-f275acdd3c92"),
+            "http://myMethodVersionUrl.com");
+    when(methodVersionDao.getMethodVersion(any())).thenReturn(myMethodVersion);
+
+    // Arrange DAO responses
+    when(runSetDao.createRunSet(any())).thenReturn(1);
+    when(methodDao.updateLastRunWithRunSet(any())).thenReturn(1);
+    when(methodVersionDao.updateLastRunWithRunSet(any())).thenReturn(1);
+    when(runSetDao.updateStateAndRunDetails(any(), any(), any(), any(), any())).thenReturn(1);
+    when(runDao.createRun(any())).thenReturn(1);
+  }
+
+  @State({"initialize run_set"})
+  public HashMap<String, String> postRunSets() throws Exception {
+    String fixedRunSetUUID = "11111111-1111-1111-1111-111111111111";
+    String fixedRunUUID = "22222222-2222-2222-2222-222222222222";
+    when(uuidSource.generateUUID())
+        .thenReturn(UUID.fromString(fixedRunSetUUID))
+        .thenReturn(UUID.fromString(fixedRunUUID));
```

Comment:
This feels like it's assuming how the state will be used rather than just setting up non-opinionated state. Is that going to be brittle? Like if we use the `initialize run_set` state to create 2 run sets in a consumer test instead of 1, will everything after the second call to generateUUID just return the same value?

Reply:
I don't think it's possible to use this state twice in the way you're suggesting, but I understand your point about it being named/identified in the wrong way. Something like your above suggestion "Exactly one run state with ID..." is probably the way to go.